### PR TITLE
Discontinuity

### PR DESCRIPTION
**Pull request by imbcmdth**
These are the changes required to support discontinuities in playlists using MSE.

### CODE REVIEW

File: src/videojs-hls.js [hunk 1]
```diff
@@ -785,9 +815,10 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
     return;
   }
 
+  // To determine how much is buffered, we need to find the buffered region we
+  // are currently playing in and measure it's length
   if (buffered && buffered.length) {
-    // assuming a single, contiguous buffer region
-    bufferedTime = tech.buffered().end(0) - tech.currentTime();
+    bufferedTime = Math.max(0, buffered.end(0) - currentTime);
```
Comment (imbcmdth) [inline src/videojs-hls.js]: We can't assume a single region any more so we have to be smarter about calculating the amount of buffered data.

Comment (dmlap) [inline src/videojs-hls.js]: Nice.

File: src/videojs-hls.js [hunk 2]
```diff
@@ -739,6 +739,35 @@ videojs.Hls.prototype.stopCheckingBuffer_ = function() {
...
+    for (i = 0;i < buffered.length; i++) {
+      if (buffered.start(i) <= currentTime &&
+          buffered.end(i) >= currentTime) {
+        return videojs.createTimeRange(buffered.start(i), buffered.end(i));
+      }
+    }
+
+    // Just return the first range if one was not found that contain
+    // the play-head
+    return videojs.createTimeRange(buffered.start(0), buffered.end(0));
```
Comment (dmlap) [inline src/videojs-hls.js]: This default seems a bit weird given the name of the method.

File: test/videojs-hls_test.js [hunk 3]
```diff
@@ -87,24 +87,24 @@ var
...
     };
-    mockTech(player.tech);
+    mockTech(player.tech_);
 
     return player;
   },
   openMediaSource = function(player) {
     // ensure the Flash tech is ready
-    player.tech.triggerReady();
+    player.tech_.triggerReady();
```
Comment (dmlap) [inline test/videojs-hls_test.js]: Not for this PR but we should just focus on the tech for these tests in the future (i.e. don't create a player at all).

**Comment by dmlap** [comment]
Other than my comment about the negative-case for `findCurrentBuffered_`, LGTM.

**Pull request merged**